Add postgres-database-migration and ghost-database skills (#101)
timescale/tiger-docs-mcp-server: skills/ghost-database/SKILL.md

---
name: ghost-database
description: |
  Use this skill to create, manage, fork, and query PostgreSQL databases using Ghost — the first database designed for agents.

  **Trigger when user asks to:**
  - Create a new PostgreSQL database quickly
  - Manage database lifecycle in agent workflows
  - Fork a database to test changes safely
  - Create a database for agents
  - Need many databases that are billed only when queried
  - Get a connection string for a Ghost database
  - Run SQL queries against a Ghost database

>   - Pause, resume, or delete Ghost databases

Reviewer: We got rid of the ability to pause databases

  - Set up the Ghost MCP server or CLI

>   **Keywords:** Ghost, ghost.build, database, create database, fork database, PostgreSQL, managed Postgres, MCP, agent database, connection string, ghost_create, ghost_fork, ghost_sql

Reviewer: Including the MCP tool names (e.g. `ghost_create`, `ghost_fork`, `ghost_sql`) in the keywords feels a little strange to me. When would users or agents ever actually be looking for those specific MCP tools by name?

license: Apache-2.0
metadata:
  author: tigerdata
---

# Ghost Database Management

Ghost is a managed PostgreSQL service designed for agents. Your account is organized into Spaces, each with unlimited databases and forks sharing a pool of compute hours (billed in 15-minute chunks when queries are executed). Free tier: 100 hours/month, 1TB storage.

When you need continuous operation, promote any database to a **dedicated instance** — a separately billed, always-on database.

Additional features:
- **CLI and MCP native:** create and query databases from the terminal or any MCP-compatible agent
- **Instant forking:** full database copies in seconds for safe experimentation

> - **Read-only access:** connect with `--read-only` for safe agentic work against production data

Reviewer: I think this is slightly misleading.

The `--read-only` flag can be passed to `ghost connect` (to get a connection string that uses read-only mode) or `ghost psql` (to connect via psql in read-only mode), but those are mostly for _humans_.

For agents, you set read-only mode with `ghost config set read_only true`. After that, all MCP tools are locked into read-only mode - i.e. queries executed with the `ghost_sql` tool are executed in read-only mode, and the MCP tools that modify resources are blocked (i.e. `ghost_delete`, `ghost_password`, `ghost_rename` are all blocked in read-only mode).

Might be worth explaining this in more detail below (it's a pretty useful feature that isn't well-documented anywhere else).

Author: can you please check the wording now?

Reviewer: Looks good! Thanks for making those changes 👍

Website: https://ghost.build

> GitHub: https://github.com/timescale/ghost

Reviewer: The repo is not public (at least not yet). Probably shouldn't link to it here.

## Installation

```bash
curl -fsSL https://install.ghost.build | sh

> ```

Reviewer: We support [other installation methods](https://github.com/timescale/ghost?tab=readme-ov-file#installation) too, such as a PowerShell script for Windows, and installing Linux packages from Packagecloud. Probably worth mentioning them for the sake of completeness.

## Getting Started

```bash
ghost login                     # Authenticate with GitHub
ghost create                    # Create a new database
ghost list                      # List all databases
ghost connect <database>        # Get connection string
```

## Core Workflows

### Create and Query a Database

```bash
# Create a database
ghost create --name my-app-db

# Run SQL directly

> ghost sql my-app-db "CREATE TABLE users (id BIGINT GENERATED ALWAYS AS IDENTITY PRIMARY KEY, email TEXT NOT NULL UNIQUE, created_at TIMESTAMPTZ NOT NULL DEFAULT now())"

Reviewer: Commands take the database ID, not the name. This example is using the name specified in the `ghost create` command above, which is incorrect.

The same is true for all other `ghost` commands, as well. Need to make sure it's correct everywhere in this doc.

# Query it
ghost sql my-app-db "SELECT * FROM users"

# Open interactive psql session
ghost psql my-app-db
```

### Fork for Safe Experimentation

Forking creates a full copy of your database in seconds — same schema, same data. Use forks to test migrations, experiment with schema changes, or let agents explore without risk to your working database.

```bash
# Fork a database
ghost fork my-app-db --name my-app-db-experiment

# Test changes on the fork
ghost sql my-app-db-experiment "ALTER TABLE users ADD COLUMN role TEXT NOT NULL DEFAULT 'user'"

# If it worked: apply to original
ghost sql my-app-db "ALTER TABLE users ADD COLUMN role TEXT NOT NULL DEFAULT 'user'"

# If it failed: delete the fork, original is untouched
ghost delete my-app-db-experiment --confirm
```

### Pause and Resume

Pause databases you're not actively using to conserve compute hours. Storage is retained.

```bash
ghost pause my-app-db
ghost resume my-app-db --wait

> ```

Reviewer: As mentioned above, we removed the `ghost pause` command.

We should probably explain when and why databases get paused instead (I think it's after 30 days of idle time now?), and just explain how to resume them with `ghost resume`.

### Inspect Schema

```bash
ghost schema my-app-db
```

Returns an LLM-optimized schema representation of all tables, columns, indexes, and constraints.

## CLI Command Reference

| Command | Description |
|---------|-------------|
| `ghost create` | Create a new database (`--name`, `--wait`, `--json`) |
| `ghost delete` | Delete a database (`--confirm` to skip prompt) |
| `ghost fork` | Fork a database (`--name`, `--wait`, `--json`) |
| `ghost connect` | Get connection string (`--read-only` for read replicas) |
| `ghost sql` | Execute SQL query (supports stdin: `cat query.sql \| ghost sql <db>`) |
| `ghost schema` | Display database schema |
| `ghost list` | List all databases (`--json`, `--yaml`) |
| `ghost status` | Show space usage |
| `ghost pause` | Pause a running database |
| `ghost resume` | Resume a paused database (`--wait`) |
| `ghost password` | Reset password (`--generate` for auto-generated) |
| `ghost rename` | Rename a database |
| `ghost logs` | View database logs |
| `ghost psql` | Open interactive psql session (`--read-only`) |

> | `ghost login` | Authenticate with GitHub OAuth (`--headless` for CI) |

Reviewer: This is not a complete list of commands. It's missing things like `ghost api-key`, `ghost config`, `ghost logout`, `ghost feedback`, `ghost mcp`, `ghost version`, and potentially others. We should probably audit this to make sure it's complete. You can easily get the full list from `ghost --help`.

## MCP Integration

> The Ghost MCP server gives agents full database lifecycle control — create, fork, query, inspect, pause, resume, and delete databases without human intervention.

Reviewer: ```suggestion
The Ghost MCP server gives agents full database lifecycle control — create, fork, query, inspect, resume, and delete databases without human intervention.
```

### Install the MCP Server

```bash
ghost mcp install               # Auto-detects your agent (Claude Code, Cursor, etc.)
```

Supports: Claude Code, Cursor, Windsurf, Codex, Gemini, VS Code, Kiro.

### MCP Tools

Once running, agents have access to these tools:

| Tool | Description |
|------|-------------|
| `ghost_create` | Create a database |
| `ghost_delete` | Delete a database |
| `ghost_fork` | Fork a database |
| `ghost_connect` | Get connection string |
| `ghost_sql` | Execute SQL query |
| `ghost_schema` | Display database schema |
| `ghost_list` | List all databases |
| `ghost_status` | Show space usage |
| `ghost_pause` | Pause a database |
| `ghost_resume` | Resume a database |
| `ghost_password` | Reset password |
| `ghost_rename` | Rename a database |
| `ghost_logs` | View logs |
| `ghost_login` | Authenticate with GitHub OAuth |
| `ghost_feedback` | Submit feedback |

> | `search_docs` | Search Ghost and Postgres documentation |

Reviewer: There is also `view_skill` (proxied from pg-aiguide)

---
name: ghost-database
description: |
  Use this skill to create, manage, fork, and query PostgreSQL databases using Ghost — the first database designed for agents.

  **Trigger when user asks to:**
  - Create a new PostgreSQL database quickly
  - Manage database lifecycle in agent workflows
  - Fork a database to test changes safely
  - Create a database for agents
  - Need many databases that are billed only when queried
  - Get a connection string for a Ghost database
  - Run SQL queries against a Ghost database
  - Resume or delete Ghost databases

>   - Set up the Ghost MCP server or CLI

Reviewer: Given that the Ghost MCP currently proxies the pg-aiguide MCP tools (including `view_skill`), are we at all concerned that this skill could create confusion for agents that are already using the Ghost MCP?

Like, if a user asks to fork a database, how will the agent decide between using this skill (which mostly recommends using `ghost` CLI commands directly, which is not really what we want if the MCP is installed) or just calling the `ghost_fork` tool? Maybe it would use the skill first and then still decide to call the MCP tool afterwards... but even in that case, using the skill would just be extra wasted tokens (the Ghost MCP is already self-describing, so if it's already installed, I don't think there's much more that this skill would help with, beyond maybe explaining some of the less well-documented features).

It seems like this skill is _mostly_ geared towards setting up Ghost the first time and understanding its capabilities (e.g. when evaluating whether to use it). But if Ghost is already installed (as MCP server), I feel like this skill could potentially be counter-productive.

Any ideas for what we should do about that? We could potentially just filter this skill out when we proxy the `view_skill` tool, but we don't do anything like that currently (we proxy the tools as-is), and in any case, it wouldn't work for pre-existing versions of Ghost that are already in the wild.

  **Keywords:** Ghost, ghost.build, database, create database, fork database, PostgreSQL, managed Postgres, MCP, agent database, connection string
license: Apache-2.0
metadata:
  author: tigerdata
---

# Ghost Database Management

Ghost is a managed PostgreSQL service designed for agents. Your account is organized into Spaces, each with unlimited databases and forks sharing a pool of compute hours (billed in 15-minute chunks when queries are executed). Free tier: 100 hours/month, 1TB storage.

When you need continuous operation, promote any database to a **dedicated instance** — a separately billed, always-on database.

Additional features:
- **CLI and MCP native:** create and query databases from the terminal or any MCP-compatible agent

> - **Instant forking:** full database copies in seconds for safe experimentation

Reviewer 2: I thought we were avoiding "instant", since few people consider 90s to be instantaneous.

- **MCP read-only mode:** `ghost config set read_only true` locks all MCP tools into read-only — SQL queries execute in read-only mode and destructive tools (`ghost_delete`, `ghost_password`, `ghost_rename`) are blocked

Website: https://ghost.build

## Installation

Multiple installation methods are provided. If you aren't sure, use the first one.

### Install Script (macOS/Linux/WSL)

```bash
curl -fsSL https://install.ghost.build | sh
```

### Install Script (Windows PowerShell)

```powershell
irm https://install.ghost.build/install.ps1 | iex
```

### Debian/Ubuntu

```bash
curl -s https://packagecloud.io/install/repositories/timescale/ghost/script.deb.sh | sudo os=any dist=any bash
sudo apt-get install ghost
```

### Red Hat/Fedora

```bash
curl -s https://packagecloud.io/install/repositories/timescale/ghost/script.rpm.sh | sudo os=rpm_any dist=rpm_any bash
sudo yum install ghost

> ```

Reviewer 2: I think this skill could benefit from some progressive disclosure, and this section in particular seems like it'd _usually_ be a waste of tokens. You only need to _install_ ghost once.

Reviewer: True. I think maybe we should try to decide whether this skill is mostly for getting Ghost set up the first time (in which case maybe listing all of the installation methods here is fine), or if we think it's useful as a continual reference for how to use Ghost. Personally, I'm kind of skeptical that it's going to be helpful once Ghost is already installed (since the MCP is self-describing anyways, and this skill would just be a conflicting source of information). See my other comment about that. I think maybe we should try to narrow in on exactly how/when we want this skill to be used, because I'm a little concerned that this skill could end up taking priority over using the Ghost MCP tools directly.

Reviewer 2: Yeah, that's fair. I agree this skill is redundant with the MCP tools, and the skill description over-promises on what it is for in a way that could be problematic.

I kinda think there should be a "choosing a db provider" skill, and then possibly an "installing ghost" sub-page.

## Getting Started

```bash
ghost login                     # Authenticate with GitHub
ghost create                    # Create a new database (returns an ID, e.g. abc123)
ghost list                      # List all databases with their IDs
ghost connect <id>              # Get connection string
```

## Core Workflows

### Create and Query a Database

```bash
# Create a database (returns an ID like abc123)
ghost create --name my-app-db

# Run SQL directly (use the database ID)
ghost sql abc123 "CREATE TABLE users (id BIGINT GENERATED ALWAYS AS IDENTITY PRIMARY KEY, email TEXT NOT NULL UNIQUE, created_at TIMESTAMPTZ NOT NULL DEFAULT now())"

# Query it
ghost sql abc123 "SELECT * FROM users"

# Open interactive psql session

> ghost psql abc123

Reviewer: We might want to say that this requires `psql` to be installed and in your `PATH`

---
name: ghost-database
description: |
  Use this skill for persistent PostgreSQL storage in agent workflows — spin up databases on demand, test schema migrations safely on forks, and pass databases as assets between agents using Ghost.

  **Trigger when user asks to:**

>   - Create a new PostgreSQL database quickly

Reviewer 2: Do we need "quickly" here?

  - Share a database snapshot so others can spin up their own copy
  - Pass a database as an input or output between agents
  - Create a database for agents
  - Need many databases that are billed only when queried
  - Set up the Ghost MCP server or CLI

  **Note:** If the Ghost MCP server is already installed, use its tools directly — this skill is for environments without the  MCP  or as a reference for what's possible.

  **Keywords:** Ghost, ghost.build, database, create database, fork database, PostgreSQL, managed Postgres, MCP, agent database, connection string
license: Apache-2.0
metadata:
  author: tigerdata
---

# Ghost Database Management

Ghost is a managed PostgreSQL service designed for agents. Your account is organized into Spaces, each with unlimited on-demand databases and forks sharing a pool of compute hours (billed in 15-minute chunks when queries are executed). Free tier: 100 hours/month, 1TB storage.

When you need continuous operation, promote any database to a **dedicated instance** — a separately billed, always-on database.

Additional features:
- **CLI and MCP native:** create and query databases from the terminal or any MCP-compatible agent

> - **Instant forking:** full database copies in seconds for safe experimentation

Reviewer 2: ```suggestion
- **Zero-copy forking:** full database copies in seconds for safe experimentation
```

- **MCP read-only mode:** `ghost config set read_only true` locks all MCP tools into read-only — SQL queries execute in read-only mode and destructive tools (`ghost_delete`, `ghost_password`, `ghost_rename`) are blocked

Website: https://ghost.build

## Installation

Multiple installation methods are provided. If you aren't sure, use the first one.

### Install Script (macOS/Linux/WSL)

```bash
curl -fsSL https://install.ghost.build | sh
```

### Install Script (Windows PowerShell)

```powershell
irm https://install.ghost.build/install.ps1 | iex
```

### Debian/Ubuntu

```bash
curl -s https://packagecloud.io/install/repositories/timescale/ghost/script.deb.sh | sudo os=any dist=any bash
sudo apt-get install ghost
```

### Red Hat/Fedora

```bash
curl -s https://packagecloud.io/install/repositories/timescale/ghost/script.rpm.sh | sudo os=rpm_any dist=rpm_any bash
sudo yum install ghost
```

## Getting Started

**CLI**
```bash
ghost login                     # Authenticate with GitHub
ghost create                    # Create a new database (returns an ID, e.g. abc123)
ghost list                      # List all databases with their IDs

> ghost connect <id>              # Get connection string

Reviewer: ```suggestion
ghost connect <name-or-id>              # Get connection string
```

```

**MCP**
```
ghost_login()                   // Authenticate with GitHub
ghost_create({ name: "my-db" }) // → returns { id: "abc123", ... }
ghost_list()                    // List all databases with their IDs

> ghost_connect({ id: "abc123" }) // Get connection string

Reviewer: ```suggestion
ghost_connect({ name_or_id: "abc123" }) // Get connection string
```

```

## Core Workflows

### Create and Query a Database

**CLI**
```bash
# Create a database (returns an ID like abc123)

> ghost create --name my-app-db

Reviewer: ```suggestion
ghost create my-app-db
```

> # Run SQL directly (use the database ID)

Reviewer: ```suggestion
# Run SQL directly
```

ghost sql abc123 "CREATE TABLE users (id BIGINT GENERATED ALWAYS AS IDENTITY PRIMARY KEY, email TEXT NOT NULL UNIQUE, created_at TIMESTAMPTZ NOT NULL DEFAULT now())"

# Query it
ghost sql abc123 "SELECT * FROM users"

# Open interactive psql session
ghost psql abc123
```

**MCP**
```
ghost_create({ name: "my-app-db" })
// → returns { id: "abc123", ... }

> ghost_sql({ id: "abc123", query: "CREATE TABLE users (id BIGINT GENERATED ALWAYS AS IDENTITY PRIMARY KEY, email TEXT NOT NULL UNIQUE, created_at TIMESTAMPTZ NOT NULL DEFAULT now())" })

Reviewer: ```suggestion
ghost_sql({ name_or_id: "abc123", query: "CREATE TABLE users (id BIGINT GENERATED ALWAYS AS IDENTITY PRIMARY KEY, email TEXT NOT NULL UNIQUE, created_at TIMESTAMPTZ NOT NULL DEFAULT now())" })
```

> ghost_sql({ id: "abc123", query: "SELECT * FROM users" })

Reviewer: ```suggestion
ghost_sql({ name_or_id: "abc123", query: "SELECT * FROM users" })
```

```

### Fork for Safe Experimentation

Forking creates a full copy of your database in seconds — same schema, same data. Use forks to test migrations, experiment with schema changes, or let agents explore without risk to your working database. You can fork a dedicated instance into an on-demand instance — useful for testing against a production copy without paying for always-on compute.

For a complete migration testing workflow using forks — including pre/post validation queries and rollback planning — see the `postgres-database-migration` skill.

**CLI**
```bash
# Fork a database (returns the fork's ID, e.g. def456)

> ghost fork abc123 --name my-app-db-experiment

Reviewer: ```suggestion
ghost fork abc123 my-app-db-experiment
```

# Test changes on the fork
ghost sql def456 "ALTER TABLE users ADD COLUMN role TEXT NOT NULL DEFAULT 'user'"

# If it worked: apply to original
ghost sql abc123 "ALTER TABLE users ADD COLUMN role TEXT NOT NULL DEFAULT 'user'"

# If it failed: delete the fork, original is untouched

> ghost delete def456 --confirm

Reviewer: Since the commands take a name _or_ ID now, you could update these to use the name from above (e.g. `my-app-db-experiment`) instead of the ID (e.g. `def456`)

```

**MCP**
```

> ghost_fork({ id: "abc123", name: "my-app-db-experiment" })

Reviewer: ```suggestion
ghost_fork({ name_or_: "abc123", name: "my-app-db-experiment" })
```

// → returns { id: "def456", ... }

> ghost_sql({ id: "def456", query: "ALTER TABLE users ADD COLUMN role TEXT NOT NULL DEFAULT 'user'" })

Reviewer: ```suggestion
ghost_sql({ name_or_id: "def456", query: "ALTER TABLE users ADD COLUMN role TEXT NOT NULL DEFAULT 'user'" })
```

// If it worked: apply to original

> ghost_sql({ id: "abc123", query: "ALTER TABLE users ADD COLUMN role TEXT NOT NULL DEFAULT 'user'" })

Reviewer: ```suggestion
ghost_sql({ name_or_id: "abc123", query: "ALTER TABLE users ADD COLUMN role TEXT NOT NULL DEFAULT 'user'" })
```

// If it failed: delete the fork, original is untouched

> ghost_delete({ id: "def456" })

Reviewer: ```suggestion
ghost_delete({ name_or_id: "def456" })
```

```

### Auto-Pause and Resume

Databases automatically pause after 30 days of idle time to conserve compute hours. Storage is retained. Resume a paused database when you need it again:

**CLI**
```bash
ghost resume abc123 --wait
```

**MCP**
```

> ghost_resume({ id: "abc123" })

Reviewer: ```suggestion
ghost_resume({ name_or_id: "abc123" })
```

```

### Inspect Schema

**CLI**
```bash
ghost schema abc123
```

**MCP**
```

> ghost_schema({ id: "abc123" })

Reviewer: ```suggestion
ghost_schema({ name_or_id: "abc123" })
```
